packages/react-shadow/src/root.tsx

```diff
@@ -5,7 +5,15 @@ import {
 import { createShadowDOMRenderer } from '@griffel/shadow-dom';
 import * as React from 'react';
 import { createProxy } from 'react-shadow';
+import {
+  REACT_SHADOW_INSERTION_BUCKET_NAME,
+  SUPPORTS_MODIFYING_ADOPTED_STYLESHEETS,
+} from './constants';
 import type { Root } from 'react-shadow';
+import type { GriffelShadowDOMRenderer } from '@griffel/shadow-dom';
+
+// Hack to get this type that is not directly exported at the moment.
+type ExtendedCSSStyleSheet = GriffelShadowDOMRenderer['adoptedStyleSheets'][0];
```

Comment:
Is it possible to create this type once and reuse it? I see it's also being used in `makeInsertionPointSheet`

Reply:
Yes. It is actually defined in `@griffel/shadow-dom` and, imo, should be exported from there but I didn't want to get this PR bogged down in getting packages aligned :)

Reply:
Since some other work needs to happen in Griffel I'm updating there to export the type.